### PR TITLE
Week 02 day 04 mod 01

### PR DESCRIPTION
**Pull request by eerwitt**
I feel ~OK with this module. I tried using the existing but they seemed to either be the OOP intro or inheritance module while the prototype stuff doesn't give much why.

I think this lesson will be some talking to start then doing most of this code live while taking questions. If it goes well I hope the intro to `call` and `apply` will help inheritance after.

@cameronjacoby let me know if this messes up any of your plans, I'm fine to scrap any/all of it.

### CODE REVIEW

File: week-02-ajax-and-js-objects/day-04/module-01/readme.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+}
+```
+
+What will happen when I run `myChaos.havoc()` without creating it again?
+
+```js
+myChaos.havoc()
+```
+
+It causes havoc now. This means that anytime I edit the prototype for a type of object it will change all the instances of that object immediately. It will also change the behaviour of new instances.
```
Comment (cameronjacoby) [inline week-02-ajax-and-js-objects/day-04/module-01/readme.md:224]: This is a great explanation - very clear. I would also mention something about how we're saving memory by storing functionality on the prototype as opposed to the constructor.

**Comment by cameronjacoby** [comment]
Overall, I think this looks great and will be a really cool exercise. My main feedback is that the instructions to "create a library using prototypes" may seem a little abstract. Two suggestions on that:
- Do a demo/walkthrough yourself of what you want them to do (you may be already planning this) and check for points of confusion before they try it on their own. [EDITED: I just read your comment again, and you said you're going to do this :+1: ]
- Maybe give a few initial challenges that have them review creating a constructor, then practice putting methods on the prototype and calling those methods. An idea could be something like a user with a log in method that checks if the password they entered is the same as `this.password`. From my experience, it takes some repetition for students to get the syntax down - it's typical for them to mix up when they need to write the word `prototype` in terms of defining and calling prototype methods.

**Comment by cameronjacoby** [comment]
Also, this doesn't mess up my plans at all :smile: 

I think your lesson is focused on prototypes "in the wild", and I ended up taking my challenges in the direction of reviewing everything OOP (with added inheritance) and having them practice modeling data.

So, I think it's is awesome that they'll get lots of different perspectives and examples!

File: week-02-ajax-and-js-objects/day-04/module-01/readme.md (hunk 1 above)
Comment (eerwitt) [inline week-02-ajax-and-js-objects/day-04/module-01/readme.md:224]: Good idea, I'll also link to the heap profiler in Google Dev Tools.

**Comment by eerwitt** [comment]
Good idea on the challenges. I've been missing the basic challenges during the lessons and need to have more. Especially as a pulse check mid lesson.

I'm making some edits and changing to WIP

**Comment by eerwitt** [comment]
@cameronjacoby updated with the practice challenges: https://github.com/sf-wdi-24/modules/tree/week-02-day-04-mod-01/week-02-ajax-and-js-objects/day-04/module-01#object-practice

File: week-02-ajax-and-js-objects/day-04/module-01/readme.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    this.fluffiness = fluffiness;
+    this.bark = function () {
+      console.log("woof woof");
+    };
+  }
+  ```
+</details>
+
+1. <details>
+  <summary>Create a `3` year old dog named `shiba` who's `very fluffy`.</summary>
```
Comment (cameronjacoby) [inline week-02-ajax-and-js-objects/day-04/module-01/readme.md:69]: :dog2:

**Pull request merged**